src/commonMain/kotlin/org/hisp/dhis/lib/expression/Expression.kt

```diff
@@ -49,7 +49,14 @@ class Expression(
      * @see .evaluate
      */
     fun evaluate(): Any? {
-        return evaluate({ _: String -> null }, ExpressionData())
+        return evaluate(ExpressionData())
+    }
+
+    /**
+     * For testing only.
+     */
+    fun evaluate(data: ExpressionData): Any? {
```

Comment:
You can add the `internal` modifier if this method should be in the public API. It is still accessible in the tests.